lib/client/twofa/u2f/u2f.go

```diff
@@ -154,13 +155,28 @@ func doU2FAuthenticate(
 	}
 	// Now we ask the token to sign/authenticate
 	logger.Println("authenticating, provide user presence")
+	retryCount := 0
 	var rawBytes []byte
 	for {
 		res, err := t.Authenticate(req)
 		if err == u2ftoken.ErrPresenceRequired {
 			time.Sleep(200 * time.Millisecond)
 			continue
 		} else if err != nil {
+			if runtime.GOOS == "darwin" && retryCount < 3 {
+				retryCount += 1
+				if err.Error() == "hid: general error" {
```

Comment:
might it make sense to add something like `|| err.Error() == "hid: privilege violation"`" to retry on the issue that is reported in https://github.com/Cloud-Foundations/keymaster/issues/2